src/main/java/com/gd/clinic/service/PatientService.java

```diff
@@ -0,0 +1,50 @@
+package com.gd.clinic.service;
+
+import com.gd.clinic.entity.Patient;
+import com.gd.clinic.repository.PatientRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.transaction.Transactional;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class PatientService {
+
+    private final PatientRepository patientRepository;
+
+    public Patient getById(UUID id) throws RuntimeException {
+        return patientRepository.findById(id).orElseThrow(() -> new RuntimeException("Patient Not Found"));
+    }
+
+    public boolean existsByInsuranceNumber(String in) {
+        return patientRepository.findOneByInsuranceNumber(in).isPresent();
+    }
+
+    public Patient save(Patient patient) {
+        patient.setCreatedBy(getCurrentUser());
+        patient.setCreatedAt(OffsetDateTime.now());
+        return patientRepository.save(patient);
+    }
+
+    public Patient getByIN(String in) {
+        return patientRepository.findOneByInsuranceNumber(in).orElseThrow(() -> new RuntimeException("Patient Not Found"));
+    }
+    public List<Patient> getAll() {
+        return patientRepository.findAll();
+    }
+
+    private String getCurrentUser() {
```

Comment:
generally, this logic is for controller layer. you need to retrieve current user there and pass it to the service
and for sure refactor as I mentioned before

also if it's used in several different classes, you can move it to some utility class

Reply:
Yes, I refactored the `getCurrentUser()` method and it is now static in its own class, `AuthInfo`. Meaning that the method can be called anywhere within the Spring Security Context. Since the `PatientService` object is instantiated in the `PatientsController`, it is in the controller layer, so in order to avoid passing unnecesary arguments to the `save()` method, the static method from the `AuthInfo` class is just called within `save()` this commit.